docs/installation/prepare/microcontrollers/_index.md

```diff
@@ -1,387 +1,390 @@
+
+#### Install Dependencies
+
+{{< tabs >}}
+{{% tab name="Linux (Ubuntu)" %}}
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+sudo apt-get install git libssl-dev dfu-util libusb-1.0-0 libudev-dev
+```
+
+{{% /tab %}}
+{{% tab name="macOS" %}}
+
+If you haven't already, install Homebrew:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+```
+
+Then, install `dfu-util`:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+brew install dfu-util
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+#### Install `Rust`
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+See the [Rust Installation guide](https://www.rust-lang.org/tools/install) for more information and other installation methods.
+
+#### Install `cargo-generate` with `cargo`
+
+ `cargo` installs automatically when downloading Rust with rustup.
+
+Run the following command to install `cargo-generate`:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cargo install cargo-generate
+```
+
+#### Install `espflash`
+
+Run the following command to install `espflash`
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cargo install espflash
+```
+
+## Set up your development enviroment
+
+{{< tabs >}}
+{{% tab name="Canon CLI (recommended)" %}}
+
+### Use Viam's Canon CLI utility
+
+[Canon](https://github.com/viamrobotics/canon) is a CLI utility for managing a Docker-based canonical environment.
+
+Canon requires a working installation of Docker Engine.
+
+[Install Docker Engine](https://docs.docker.com/engine/install/) before proceeding.
+If you are running Docker Engine on Linux, make sure that you go through the [post installation steps](https://docs.docker.com/engine/install/linux-postinstall/) to run Docker automatically on startup.
+
+{{< tabs >}}
+{{% tab name="Linux (Ubuntu)" %}}
+
+Make sure your system has [Go 1.19](https://golangtutorial.dev/news/go-1.19-version-released/#major-changes-in-go-119-version) or later installed.
+Verify your version of Go with `go version`.
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+go install github.com/viamrobotics/canon@latest
+```
+
+{{% /tab %}}
+{{% tab name="macOS" %}}
+
+Make sure your system has [Go 1.19](https://golangtutorial.dev/news/go-1.19-version-released/#major-changes-in-go-119-version) or later installed.
+Verify your version of Go with `go version`.
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+brew install viamrobotics/brews/canon
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+Make sure to [add the go binary folder to your `PATH`](https://go.dev/doc/gopath_code) by running:
+`export PATH=$PATH:$(go env GOPATH)/bin`.
```

Comment:
This is only needed for the "go install" installation method, and not needed for homebrew (installing homebrew will have already set the user up with the correct PATH)

Reply:
Given that each linux distribution has its own package manager I don't want users to be forced to use Homebrew. Installing go via the command line seems the least invasive and relatively easy to revert if something goes wrong, the user would have to figure out how to install golang for their particular distro